docs/release_process.md

```diff
@@ -0,0 +1,277 @@
+
+```mermaid
+%%{init: { 'theme': 'base', 'gitGraph': {'rotateCommitLabel': true} } }%%
+    gitGraph
+       commit
+       branch change-1
+       commit
+       commit
+       checkout main
+       branch change-2
+       commit
+       checkout main
+       merge change-1
+       branch change-3
+       commit
+       commit
+       checkout main
+       merge change-2 tag: "2022.08.02"
+       checkout change-3
+       commit
+       commit
+       checkout main
+```
+
+## Release to Prod
+We leave the tag on [Stage][stage] for a week so that we (and especially QA)
+can check the tag on GCP infrastucture before we deploy it to production. To
+deploy the tag to production:
+
+1. [Make a release on GitHub][github-new-release] for the tag.
+   * Use the "Generate release notes" button!
+   * Mark it as a pre-release.
+2. File an [SRE ticket][sre-board] to deploy the tag to [Prod][prod].
+   * Include a link to the GitHub Release
+   * When the tag goes to prod, update the GitHub Release from "pre-release" to
+     a full release.
+
+## Stage-fixes
+Ideally, every change can ride the regular weekly release "trains". But
+sometimes we need to make and release changes before the regularly scheduled
+release.
+
+### "Clean `main`" flow
+If a bug is caught on [Stage][stage] in a tag that is scheduled to go to
+[Prod][prod], we need to fix the bug before the scheduled prod deploy. If
+`main` is "clean" - i.e., nothing else has merged yet, we can use the regular
+GitHub Flow:
+
+1. Create a stage-fix branch from the tag. E.g.:
+   * `git branch stage-fix-2022.08.02 2022.08.02`
+   * `git switch stage-fix-2022.08.02`
+2. Make changes
+3. Create a pull request to `main`
+4. Address review comments
+5. Merge pull request
+6. Make and push a new tag. E.g.: `2022.08.02.01
+
+```mermaid
+%%{init: { 'theme': 'base' } }%%
+    gitGraph
+       commit
+       branch change-1
+       commit
+       commit
+       checkout main
+       branch change-2
+       commit
+       checkout main
+       merge change-1
+       branch change-3
+       commit
+       commit
+       checkout main
+       merge change-2 tag: "2022.08.02"
+       checkout change-3
+       commit
+       commit
+       checkout main
+       branch stage-fix-2022.08.02
+       commit
+       checkout main
+       merge stage-fix-2022.08.02 tag: "2022.08.02.01"
+```
+
+### "Dirty `main`" flow
+If a bug is caught on [Stage][stage] in a tag that is scheduled to go to
+[Prod][prod], we need to fix the bug before the scheduled prod deploy. If
+`main` is "dirty" - i.e., other changes have merged, we can make the new tag
+from the stage-fix branch.
+
+1. Create a stage-fix branch from the tag. E.g.:
+   * `git branch stage-fix-2022.08.02 2022.08.02`
+   * `git switch stage-fix-2022.08.02`
+2. Make changes
+3. Create a pull request to `main`
+4. Address review comments
+5. Merge pull request
+6. Make and push a new tag *from the `stage-fix` branch*
```

Comment:
Hmm ... it looks like it *will* generate release notes, but it will only be the "Full Changelog" link, instead of the nicer-looking list of PRs merged. The full changelog seems like it's fine for "dirty main" point releases. It's the "dirty" flow, after all.

<img width="1033" alt="image" src="https://user-images.githubusercontent.com/71928/184377631-37ce2bb1-6cf3-4453-abba-8162259650ad.png">


<img width="1031" alt="image" src="https://user-images.githubusercontent.com/71928/184377719-d683c113-a025-4a9a-959b-692ac54c5196.png">


<img width="908" alt="image" src="https://user-images.githubusercontent.com/71928/184378384-6bea8ea2-611b-4b6a-a2bc-008486ae95f1.png">

Reply:
I updated the doc to include steps for deploying to prod, checking prod deploy, and making GitHub release notes - based on yesterday's deployment steps and our Slack thread.